zdict/dictionaries/oxford.py

```diff
@@ -0,0 +1,192 @@
+                else:
+                    print()
+
+                # entry
+                idx = 1
+                for entry in lex_ent['entries']:
+                    for sense in entry['senses']:
+                        line_prefix = '{idx}.'.format(idx=idx)
+                        self._show_sense(sense, line_prefix)
+                        idx += 1
+
+        print()
+
+    def _show_sense(self, sense: dict, prefix='', indent=1):
+        print()
+        self.color.print(prefix, end=' ', indent=indent)
+
+        # regions
+        if 'regions' in sense:
+            regions_str = ', '.join(sense['regions'])
+            regions_str = '(' + regions_str + ')'
+            self.color.print(regions_str, 'yellow', end=' ')
+
+        # register
+        if 'registers' in sense:
+            registers_str = ', '.join(sense['registers'])
+            self.color.print(registers_str, 'red', end=' ')
+
+        # domain
+        if 'domains' in sense:
+            domains_str = ', '.join(sense['domains'])
+            domains_str = '(' + domains_str + ') '
+            self.color.print(domains_str, 'green', end='')
+
+        # notes
+        if 'notes' in sense:
+            notes = [note['text'] for note in sense['notes']]
+            notes_str = ', '.join(notes)
+            notes_str = '[' + notes_str + '] '
+            self.color.print(notes_str, 'magenta', end='')
+
+        # definition
+        if 'definitions' in sense:
+            definition_str = '. '.join(sense['definitions'])
+            print(definition_str)
+
+        # cross ref
+        if 'crossReferenceMarkers' in sense:
+            xref_marker_str = '. '.join(sense['crossReferenceMarkers'])
+            print(xref_marker_str)
+
+        # example
+        for example in sense.get('examples', []):
+            print()
+            print(' ' * (indent + 1), end='  ')
+            self.color.print(example['text'], 'indigo', indent=indent+2)
+
+        # subsenses
+        if self.args.verbose and 'subsenses' in sense:
+            for idx, subsense in enumerate(sense['subsenses'], 1):
+                line_prefix = '{prefix}{idx}.'.format(prefix=prefix, idx=idx)
+                self._show_sense(subsense, line_prefix, indent=indent + 1)
+
+    def _get_app_key(self):
+        """
+        Get the app id & key for query
+
+        .. note:: app key storage
+            The API key should placed in ``KEY_FILE`` in the ``~/.zdict`` with
+            the format::
+
+                app_id,app_key
+
+        .. note:: request limit
+            request limit: per minute is 60, per month is 3000.
+        """
+        if not os.path.exists(self.KEY_FILE):
+            self._show_instruction()
+            raise APIKeyError('Oxford: API key not found')
+
+        with open(self.KEY_FILE) as fp:
+            keys = fp.read()
+
+        keys = re.sub('\s', '', keys).split(',')
+        if len(keys) != 2:
+            self._show_instruction()
```

Comment:
I think we don't need to use `self._show_instruction()` here because users might already store the key in `KEYFILE` but not with correct format. Just remind users for the correct format of the key would be enough.